solc/CommandLineInterface.cpp

```diff
@@ -1087,61 +1099,91 @@ bool CommandLineInterface::assemble(
 	return true;
 }
 
+bool CommandLineInterface::assembleFromAssemblyJson()
+{
+	solAssert(m_fileReader.sourceCodes().size() == 1, "");
+
+	Json::Value assemblyJson;
+	m_assembly = make_unique<evmasm::Assembly>();
+	return util::jsonParseStrict(m_fileReader.sourceCodes().begin()->second, assemblyJson) &&
+				m_assembly->loadFromAssemblyJSON(assemblyJson, *m_assembly);
```

Comment:
I think it would be better not to create an empty assembly if the function fails.